### PR TITLE
Adds parser options

### PR DESCRIPTION
**Pull request by jaguirresza**
Adds parser options, these are just passed directly to cheerio.

### CODE REVIEW

**Comment by jorgeaguirre** [comment]
Because not all backends are well implemented, and in some cases cookies that were encoded with `encodeURIComponent` are not recognised, thus becomes necessary the use another encoding function.

**Comment by jaguirresza** [comment]
i will fix it and create a new pull request

File: gurkha.js [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+                    'object, string or array');
+  }
+
+  if (options !== undefined && typeof(schema) !== 'object') {
+    throw new Error('Illegal argument: if options are present must ' +
+                    'be an object.');
   }
 
   this._schema = schema;
+  this._options = options || {};
```
Comment (jorgeaguirre) [inline gurkha.js]: How can i fix it?

**Comment by jorgeaguirre** [comment]
How can i fix the 'possible strict violation'

File: gurkha.js [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
 
-function gurkha (schema) {
+function gurkha (schema, options) {
   if (typeof(schema) !== 'object' && typeof(schema) !== 'string') {
-    throw new Error('Illegal argument: constructor must receive a schema object, string or array');
+    throw new Error('Illegal argument: constructor must receive a schema' +
+                    'object, string or array');
+  }
+
+  if (options !== undefined && typeof(schema) !== 'object') {
```
Comment (monitz87) [inline gurkha.js]: Shouldn't this be

``` javascript
 if (options !== undefined && typeof(options) !== 'object') 
```

**Comment by monitz87** [comment]
You can't, I need to update the linter.

Just ignore hound for the moment

**Pull request merged**